public/app/features/auth-config/state/actions.ts

```diff
@@ -0,0 +1,71 @@
+import { lastValueFrom } from 'rxjs';
+
+import { getBackendSrv, isFetchError } from '@grafana/runtime';
+import { contextSrv } from 'app/core/core';
+import {
+  AccessControlAction,
+  Settings,
+  ThunkResult,
+  SettingsError,
+  UpdateSettingsQuery,
+  AuthProviderStatus,
+} from 'app/types';
+
+import { getAuthProviderStatus, getRegisteredAuthProviders } from '..';
+
+import { providerStatusesLoaded, resetError, setError, settingsUpdated } from './reducers';
+
+export function loadSettings(): ThunkResult<Promise<Settings>> {
+  return async (dispatch) => {
+    if (contextSrv.hasPermission(AccessControlAction.SettingsRead)) {
+      const result = await getBackendSrv().get('/api/admin/settings');
+      dispatch(settingsUpdated(result));
+      dispatch(loadProviderStatuses());
+      return result;
+    }
+  };
+}
+
+export function loadProviderStatuses(): ThunkResult<void> {
+  return async (dispatch) => {
+    const registeredProviders = getRegisteredAuthProviders();
+    const providerStatuses: { [key: string]: AuthProviderStatus } = {};
+    for (const provider of registeredProviders) {
+      const status = await getAuthProviderStatus(provider.id);
+      providerStatuses[provider.id] = status;
```

Comment:
Wonder if building an array of promises and using `Promise.all` will be more efficient here.